src/cobra/core/gene.py

```diff
@@ -88,29 +67,38 @@ def eval_gpr(expr, knockouts):
         True if the gene reaction rule is true with the given knockouts
         otherwise false
     """
-    if isinstance(expr, Expression):
-        return eval_gpr(expr.body, knockouts)
-    elif isinstance(expr, Name):
-        return expr.id not in knockouts
-    elif isinstance(expr, BoolOp):
-        op = expr.op
-        if isinstance(op, Or):
-            return any(eval_gpr(i, knockouts) for i in expr.values)
-        elif isinstance(op, And):
-            return all(eval_gpr(i, knockouts) for i in expr.values)
-        else:
-            raise TypeError("unsupported operation " + op.__class__.__name__)
-    elif expr is None:
-        return True
-    else:
-        raise TypeError("unsupported operation  " + repr(expr))
+    warn(
+        "eval_gpr() will be removed soon." "Use GPR().eval(knockouts) in the future",
+        DeprecationWarning,
+    )
+    return expr.eval(knockouts=knockouts)
```

Comment:
Probably fails if `expr` is not an instance of `GPR`, for instance BoolOp. I think you meant `return GPR(expr).eval(knockouts=knockouts)`.